src/Core/Containers/DynamicVisitor.hpp

```diff
@@ -0,0 +1,163 @@
+    using OperatorsStorageType = std::unordered_map<std::type_index, CallbackFunction>;
+
+    /// \brief Helper struct to build type-erased operators (allowing to test calling profile)
+    template <typename T, typename F, bool WithUserParam>
+    struct MakeVisitOperatorHelper {
+        auto makeOperator( F& f ) -> OperatorsStorageType::value_type;
+    };
+
+    /// \brief construct a type-erased operator from a user define functor with profile
+    /// void(const std::string, [const]T[&] [, [const] std::any&&])
+    template <typename T, typename F>
+    auto makeVisitorOperator( F& f ) -> OperatorsStorageType::value_type;
+
+    /// \brief The type erased operators
+    OperatorsStorageType m_visitorOperator;
+};
+
+template <typename T, typename F>
+bool DynamicVisitor::addOperator( F&& f ) {
+    auto [it, inserted] = m_visitorOperator.insert( makeVisitorOperator<T, F>( f ) );
+    return inserted;
+}
+
+template <typename T>
+bool DynamicVisitor::hasOperator() {
+    return m_visitorOperator.find( VariableSet::getVariableVisitTypeIndex<T>() ) !=
+           m_visitorOperator.end();
+}
+
+template <typename T, typename F>
+void DynamicVisitor::addOrReplaceOperator( F&& f ) {
+    auto op = makeVisitorOperator<T, F>( f );
+    m_visitorOperator.insert_or_assign( op.first, op.second );
+}
+
+template <typename T>
+bool DynamicVisitor::removeOperator() {
+    assert( hasOperator<T>() );
+    auto res = m_visitorOperator.erase( VariableSet::getVariableVisitTypeIndex<T>() ) > 0;
+    return res;
+}
+
+template <typename T, typename F>
+struct DynamicVisitor::MakeVisitOperatorHelper<T, F, true> {
+    inline auto makeOperator( F& f ) -> OperatorsStorageType::value_type {
+        return { VariableSet::getVariableVisitTypeIndex<T>(),
+                 [&f]( const std::string& name, std::any& a, std::any&& userParam ) {
+                     auto rp = std::any_cast<std::reference_wrapper<T>>( a );
+                     auto& p = rp.get();
+                     f( name, p, std::forward<std::any>( userParam ) );
+                 } };
+    }
+};
+
+template <typename T, typename F>
+struct DynamicVisitor::MakeVisitOperatorHelper<T, F, false> {
+    inline auto makeOperator( F& f ) -> OperatorsStorageType::value_type {
+        return { // type index
+                 VariableSet::getVariableVisitTypeIndex<T>(),
+                 // callback
+                 [&f]( const std::string& name, std::any& a, std::any&& ) {
+                     auto rp = std::any_cast<std::reference_wrapper<T>>( a );
+                     auto& p = rp.get();
+                     f( name, p );
+                 } };
+    }
+};
+
+template <class T, class F>
+inline auto DynamicVisitor::makeVisitorOperator( F& f ) -> OperatorsStorageType::value_type {
+    auto opBuilder = MakeVisitOperatorHelper < T, F,
+         std::is_invocable<F, const std::string&, T&, std::any&&>::value ||
+             std::is_invocable<F, const std::string&, T&, std::any&&>::value > {};
```

Comment:
it seems the two arguments of the `||` operator are the same.

Reply:
I don't remember how/why I added `&` to the first T, compiles and passes tests without it. It happens when I move this code to own source file. Since I'm not sure what is expected... is it
```suggestion
```
or
```suggestion
             std::is_invocable<F, const std::string&, T, std::any&&>::value > {};
```

Reply:
From line 891 of the actual src/Core/Containers/VariableSet.hpp, it is the second. This test for a callable with userParam(3rd parameter)  with the second parameter pass as value or as reference. This is to relate to the partial specialization of the struct `struct DynamicVisitor::MakeVisitOperatorHelper<T, F, [true | false]>` some lines above

Reply:
My bad, it's commit 75d69781d7c2a3c0ae84e8eb8e0db8dc17219611 which introduce this modification. 
1. Compiles and pass tests
```
std::is_invocable<F,` const std::string&, T&, std::any&&>::value
```

2. Does not compile
```
std::is_invocable<F,` const std::string&, T, std::any&&>::value
```

3. Compiles and pass tests.
```
std::is_invocable<F, const std::string&, T, std::any&&>::value ||
std::is_invocable<F, const std::string&, T&, std::any&&>::value
```


So it seems that we haven't a test that require non ref `T`. If such case is not needed, it's ok to remove the first part of the "or" (and go to case 1.). But if there is any case one need to have both type of functor, we need to add the corresponding test and go to case 3.

Reply:
I removed non ref `T` version and also moved to enable_if instead of helper struct.